packages/desktop-client/src/components/mobile/MobileAmountInput.js

```diff
@@ -0,0 +1,399 @@
+            top: 0,
+
+            backgroundColor: animationColor || colors.p10,
+            opacity: this.backgroundValue,
+            borderRadius: 2,
+          }}
+          pointerEvents="none"
+        /> */}
+        <Text
+          style={textStyle}
+          data-testid="amount-fake-input"
+          pointerEvents="none"
+        >
+          {editing ? text : amountToCurrency(value)}
+        </Text>
+      </View>
+    );
+  }
+}
+
+const AmountInputWithContext = forwardRef((props, ref) => {
+  return (
+    <AmountAccessoryContext.Consumer>
+      {context => <AmountInput {...props} ref={ref} context={context} />}
+    </AmountAccessoryContext.Consumer>
+  );
+});
+
+class FocusableAmountInput_ extends PureComponent {
+  state = { focused: false, isNegative: true };
+
+  componentDidMount() {
+    const onDone = () => {
+      this.setState({ focused: false });
+      // Keyboard.dismiss();
+    };
+    this.props.context.on('done', onDone);
+    this.cleanup = () => {
+      this.props.context.off('done', onDone);
+    };
+
+    if (this.props.sign) {
+      this.setState({ isNegative: this.props.sign === 'negative' });
+    } else if (
+      this.props.value > 0 ||
+      (!this.props.zeroIsNegative && this.props.value === 0)
+    ) {
+      this.setState({ isNegative: false });
+    }
+  }
+
+  componentWillUnmount() {
+    this.cleanup();
+  }
+
+  focus = () => {
+    this.setState({ focused: true });
+  };
+
+  onFocus = () => {
+    this.focus();
+  };
+
+  toggleIsNegative = () => {
+    this.setState({ isNegative: !this.state.isNegative }, () => {
+      this.onBlur(this.props.value);
+    });
+  };
+
+  onBlur = value => {
+    this.setState({ focused: false, reallyFocused: false });
+    if (this.props.onBlur) {
+      const absValue = Math.abs(value);
+      this.props.onBlur(this.state.isNegative ? -absValue : absValue);
+    }
+  };
+
+  render() {
+    const { textStyle, style, focusedStyle, buttonProps } = this.props;
+    const { focused } = this.state;
+
+    return (
+      <View>
+        <AmountInputWithContext
+          {...this.props}
+          ref={el => (this.amount = el)}
+          onBlur={this.onBlur}
+          focused={focused}
+          style={[
+            {
+              width: 80,
+              transform: [{ translateX: 6 }],
+              justifyContent: 'center',
+            },
+            style,
+            focusedStyle,
+            !focused && {
+              opacity: 0,
+              position: 'absolute',
+              top: 0,
+            },
+          ]}
+          textStyle={[{ fontSize: 15, textAlign: 'right' }, textStyle]}
+        />
+
+        <View>
+          {!focused && (
+            <Button
+              style={{
+                position: 'absolute',
+                right: 'calc(100% + 5px)',
+                top: '8px',
+              }}
+              onClick={this.toggleIsNegative}
+            >
+              {this.state.isNegative ? '-' : '+'}
```

Comment:
`+` and `-` are slightly different widths which makes the button move slightly when tapped. It might be worth trying the real minus sign `−` (`\u{2212}`) to see if it’s the same width as `+`. If that doesn’t work maybe add a fixed width?

Reply:
Good call! The actual minus sign seems to be the same width and stops the button from moving. Done via https://github.com/actualbudget/actual/pull/1340/commits/2b5f39039c66858dde33171ef6c1846b6434f384.